src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -149,6 +156,32 @@ public PersistentDataStore() {
     return messages;
   }
 
+  public HashMap<String, Hashtag> loadHashtags() throws PersistentDataStoreException {
+
+    HashMap<String, Hashtag> hashtags = new HashMap<String, Hashtag>();
+
+    // Retrieve all hashtags from the datastore.
+    Query query = new Query("chat-hashtags").addSort("creation_time", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
```

Comment:
You need to change line 168-171 to match other loads like (loadMessages). You CANNOT just retrieve the contents and leave out the IDs, creation time, list of users, lists of conversations.